src/kademlia/node.cpp

```diff
@@ -572,14 +572,22 @@ void node::send_single_refresh(udp::endpoint const& ep, int bucket
 	e["y"] = "q";
 	entry& a = e["a"];
 
-	// use get_peers instead of find_node. We'll get nodes in the response
-	// either way.
-	e["q"] = "get_peers";
-	a["info_hash"] = target.to_string();
-	m_counters.inc_stats_counter(counters::dht_get_peers_out);
-
-//	e["q"] = "find_node";
-//	a["target"] = target.to_string();
+	if (m_table.bucket_size(bucket) < m_table.bucket_limit(bucket)
+		|| m_table.replacements_size(bucket) < m_table.bucket_size())
```

Comment:
would it make sense to move this logic into the routing table and just have a function: routing_table::is_full(bucket)?

it seems like if you have to expose a new function to query the number of replacement nodes, you might as well just make the routing_table able to tell you if it's full. (basically, having all these calls into the routing table and compare the output seems like a bit of a red-flag that the interface to the class is cut at the wrong place)